pkg/reconciler/workload/basecontroller/controller.go

```diff
@@ -246,20 +260,23 @@ func (c *ClusterReconciler) process(ctx context.Context, key string) error {
 }
 
 func (c *ClusterReconciler) deletedCluster(obj interface{}) {
+	logger := logging.WithReconciler(klog.Background(), c.name)
```

Comment:
didn't we choose to make logger an arg of event handlers?

Reply:
Not that I was aware of, no - in one controller that Andy had written there was a lot of nested calls to event handlers with each call adding context so having one made sense. We can certainly refactor more broadly as a follow-up though.